iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java

```diff
@@ -123,8 +123,14 @@ public double scale(Network n, double asked, ScalingParameters parameters) {
         Terminal t = g.getTerminal();
         if (!t.isConnected()) {
             if (parameters.isReconnect()) {
-                new ConnectGenerator(g.getId()).apply(n);
                 LOGGER.info("Connecting {}", g.getId());
+                new ConnectGenerator(g.getId()).apply(n);
+                if (!t.isConnected() || !g.getTerminal().getBusBreakerView().getConnectableBus().isInMainConnectedComponent()) {
+                    //If the generator still disconnected, we should not change the active power setPoint
+                    LOGGER.info("Generator {} could not be connected, discarded from scaling", g.getId());
+                    return 0.;
+                }
+
```

Comment:
Please extract this part and the few surrounding lines in a new method to reduce the method complexity and avoid a brain overload issue